modules/veteran/app/workers/veteran/vso_reloader.rb

```diff
@@ -3,54 +3,58 @@
 require 'sidekiq'
 
 module Veteran
-  class VsoReloader
-    include Sidekiq::Worker
-    BASE_URL = 'https://www.va.gov/ogc/apps/accreditation/'
-
+  class VsoReloader < BaseReloader
     def perform
       array_of_organizations = reload_representatives
-      Veteran::Service::Organization.import(array_of_organizations, on_duplicate_key_ignore: true)
+      Veteran::Service::Representative.where.not(representative_id: array_of_organizations).destroy_all
     end
 
     private
 
     def reload_representatives
-      data = fetch_data('orgsexcellist.asp')
-      data.each do |hash|
-        find_or_create_representative(hash)
+      vso_reps = []
+      vso_orgs = fetch_data('orgsexcellist.asp').map do |hash|
+        find_or_create_vso(hash)
+        vso_reps << hash['Registration Num']
+        { poa: hash['POA'], name: hash['Organization Name'], phone: hash['Org Phone'], state: hash['Org State'] }
+      end
+      Veteran::Service::Organization.import(vso_orgs, on_duplicate_key_ignore: true)
+
+      attorneys = fetch_data('attorneyexcellist.asp').map do |hash|
+        find_or_create_attorneys(hash)
+        hash['Registration Num']
       end
 
-      Veteran::Service::Representative.where.not(representative_id: data.map { |h| h['Registration Num'] }).destroy_all
+      claim_agents = fetch_data('caexcellist.asp').map do |hash|
+        find_or_create_claim_agents(hash)
+        hash['Registration Num']
+      end
 
-      array_of_organizations = data.map do |h|
-        { poa: h['POA'], name: h['Organization Name'], phone: h['Org Phone'], state: h['Org State'] }
-      end.uniq.compact
-      array_of_organizations
+      vso_reps + attorneys + claim_agents
+    end
+
+    def find_or_create_attorneys(hash)
+      rep = find_or_initialize(hash)
+      rep.user_types << 'attorney'
+      rep.save
     end
 
-    def find_or_create_representative(hash)
-      rep = Veteran::Service::Representative.find_or_initialize_by(representative_id: hash['Registration Num'])
+    def find_or_create_claim_agents(hash)
+      rep = find_or_initialize(hash)
+      rep.user_types << 'claim_agents'
```

Comment:
same as above

Reply:
Checking now to make sure it doesn't do double entry.